Rock.Messaging.SQS/MQ/SQSQueueReceiver.cs

```diff
@@ -103,71 +106,83 @@ private void DoStuff()
                     continue;
                 }
 
-                foreach (var message in response.Messages)
+                if (_parallelHandling)
                 {
-                    if (_stopped)
+                    Parallel.ForEach(response.Messages, Handle);
+                }
+                else
+                {
+                    foreach (var message in response.Messages)
                     {
-                        break;
+                        Handle(message);
                     }
+                }
+            }
+        }
+
+        private void Handle(Message message)
+        {
+            if (_stopped)
+            {
+                return;
+            }
 
-                    var handler = MessageReceived;
+            var handler = MessageReceived;
+
+            if (handler != null)
+            {
+                var receiptHandle = message.ReceiptHandle;
 
-                    if (handler != null)
+                Action acknowledge =
+                    () =>
                     {
-                        var receiptHandle = message.ReceiptHandle;
+                        Exception deleteException = null;
+                        DeleteMessageResponse deleteResponse = null;
 
-                        Action acknowledge =
-                            () =>
+                        for (int i = 0; i < 3; i++)
```

Comment:
New to this code, but what is '3'?  Seems like a magic number.  Could this be better described as a Const?

Reply:
This was just moved around, but I can make it a const 😄

Reply:
I looked around and was not able to understand what 3 meant.  So making it a const would help the next dev.

Reply:
Not 100% sure I pegged the motivations right, as I didn't write this loop, but I took a crack at interpreting this as a const (as well as the other loop bound).

Reply:
It's a simple retry loop. If the call to aws fails, try again two more times.